### PR TITLE
System to mount Nextflow Tower configuration

### PR DESCRIPTION
**Pull request by tthyer**
- Add an S3 bucket that will be populated with configuration file(s)
- Add an EFS file system that will be mounted into Tower docker containers to obtain configuration
- Add DataSync task that will sync the S3 bucket file(s) to the EFS file system

[WORKFLOWS-38](https://sagebionetworks.jira.com/browse/WORKFLOWS-38)

### CODE REVIEW

File: templates/nextflow-ecs-task-definition.yaml
Comment (zaro0508) [inline templates/nextflow-ecs-task-definition.yaml:120]: `efs_volume` string appears a few times in this file.  how about making it a parameter?

File: templates/nextflow-tower-config-bucket.yaml
Comment (zaro0508) [inline templates/nextflow-tower-config-bucket.yaml:43]: your passing in SSO administrators and ci service account for AdminArns. those users/roles already have full bucket access so I'm not sure why you need to give them access again?

**Review by zaro0508 — CHANGES_REQUESTED** [review]

File: config/develop/nextflow-efs-file-system.yaml
Comment (tthyer) [reply config/develop/nextflow-efs-file-system.yaml:7]: Gotcha, I've been a little more explicit in this repo since there are so many interdependencies, and I want to call that out.

File: config/develop/nextflow-tower-config-bucket.yaml
Comment (tthyer) [reply config/develop/nextflow-tower-config-bucket.yaml:1]: You don't! Cool huh?

File: config/develop/nextflow-tower-config-bucket.yaml
Comment (tthyer) [reply config/develop/nextflow-tower-config-bucket.yaml:11]: It is completely fine for that to happen here.

File: templates/nextflow-ecs-task-definition.yaml
Comment (tthyer) [reply templates/nextflow-ecs-task-definition.yaml]: parameterized in c4506b4

File: templates/nextflow-tower-config-bucket.yaml
Comment (tthyer) [reply templates/nextflow-tower-config-bucket.yaml]: Good point -- removed in eef71a1

File: config/develop/nextflow-tower-config-bucket.yaml
Comment (tthyer) [reply config/develop/nextflow-tower-config-bucket.yaml]: As it happens, `AdminArns` were removed, but will definitely be using that stack_group_config field elsewhere

File: templates/nextflow-tower-config-bucket.yaml
Comment (tthyer) [reply templates/nextflow-tower-config-bucket.yaml:20]: presumably this is now relevant

File: config/develop/nextflow-tower-config-bucket.yaml
Comment (zaro0508) [reply config/develop/nextflow-tower-config-bucket.yaml:1]: i guess that's because `#` isn't a comment in jinja however it looks like a comment because it's a .yaml file.  anyways might be nice to uncomment for readability but your call.

File: templates/nextflow-efs-file-system.yaml
Comment (zaro0508) [reply templates/nextflow-efs-file-system.yaml:30]: not gonna do anything about this one?

File: config/develop/nextflow-tower-config-bucket.yaml
Comment (tthyer) [reply config/develop/nextflow-tower-config-bucket.yaml]: @zaro0508 Sadly, in order for this to stay a yaml file I would have to disable yamllint for the whole file. I thought that using the comment trick was a convenient workaround. Instead I have converted this into a j2 file. I don't really like that solution either because now I'm not getting the benefit of linting either. What do you normally do so that you get linted but can add jinja statements?

File: templates/nextflow-efs-file-system.yaml
Comment (tthyer) [reply templates/nextflow-efs-file-system.yaml:30]: @zaro0508 Do you think I need to encrypt a bucket that only contains config files for Tower, i.e. not PHI?

File: templates/nextflow-tower-config-bucket.yaml
Comment (tthyer) [reply templates/nextflow-tower-config-bucket.yaml:20]: since there is cross-account access happening, that is

**Review by BrunoGrandePhD — APPROVED** [review]

**Comment by tthyer** [comment]
@zaro0508 I have to deploy these changes in order to unblock other work, but if you have more comments I can address them in a follow-up PR

File: templates/nextflow-efs-file-system.yaml
Comment (zaro0508) [reply templates/nextflow-efs-file-system.yaml:30]: i guess not but @ahayden might think otherwise.  anyways it would be pretty easy just to set encryption using the  default key correct?  i think we only require AES encryption on s3 buckets so i'm guessing default key here is fine.

File: config/develop/nextflow-tower-config-bucket.yaml
Comment (zaro0508) [reply config/develop/nextflow-tower-config-bucket.yaml]: ahh, now that i know the intention behind the comment then i guess the comment is better than converting file to j2 to bypass the linter.  The other option you can do is to keep as j2 but also enable a jinja linter, here's how to do it with github actions https://github.com/Sage-Bionetworks-IT/organizations-infra/blob/master/.github/workflows/aws-deploy.yaml#L19

File: templates/nextflow-efs-file-system.yaml
Comment (tthyer) [reply templates/nextflow-efs-file-system.yaml:30]: @zaro0508 Ok, will do this in a follow-up PR later today

File: config/develop/nextflow-tower-config-bucket.yaml
Comment (tthyer) [reply config/develop/nextflow-tower-config-bucket.yaml]: @zaro0508 In that case I'll go back to my comment. I would actually like having a jinja linter in place but it should go into pre-commit, not in the github action.

**Pull request merged**
